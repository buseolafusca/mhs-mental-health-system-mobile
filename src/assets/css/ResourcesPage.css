.expander{
	background-color: #fff;
	font-size: 1.5em;
}

.expander .nhsuk-details__summary {
	padding: 16px;
	background-color: #fff;
	display: block
}

@media (min-width:40.0625em) {
	.expander .nhsuk-details__summary {
		padding: 24px
	}
}

.expander .nhsuk-details__summary:before {
	display: none!important
}

.expander .nhsuk-details__summary:focus {
	box-shadow: 0 0 0 4px #ffb81c
}

.expander .nhsuk-details__summary:hover {
	box-shadow: 0 0 0 4px #ffcd60
}

.expander .nhsuk-details__summary:active,
.expander .nhsuk-details__summary:focus,
.expander .nhsuk-details__summary:hover {
	outline: 0
}

.expander .nhsuk-details__summary-text {
	background: url("data:image/svg+xml;charset=utf-8,%3Csvg class='nhsuk-icon nhsuk-icon__plus' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24' width='32' aria-hidden='true'%3E%3Ccircle cx='12' cy='12' r='10' fill='%23005eb8'/%3E%3Cpath fill='none' stroke='%23fff' stroke-linecap='round' stroke-miterlimit='10' stroke-width='2' d='M12 8v8m-4-4h8'/%3E%3C/svg%3E") left -2px center no-repeat;
	color: #005eb8;
	cursor: pointer;
	display: block;
	padding: 8px 8px 8px 38px;
	position: relative
}

.expander .nhsuk-details__text {
	padding: 16px;
	border-left: 0;
	margin-left: 0;
	margin-top: 0
}

@media (min-width:40.0625em) {
	.expander .nhsuk-details__text {
		padding: 24px
	}
}

.expander-group {
	margin-bottom: 16px
}

.expander-group>.nhsuk-details {
	margin-bottom: 8px
}

@media (min-width:40.0625em) {
	.expander-group>.nhsuk-details {
		margin-bottom: 8px
	}
}

@media (min-width:40.0625em) {
	.expander-group {
		margin-bottom: 24px
	}
}

.resourcepage-back-link{
	position: relative;
    z-index: 1;
    margin: 5%;
}

.resourcepage-back-link__link{
    font-size: 0.8em;
}

.resourcepage-icon__chevron-left{
	position: relative;
}

.place-title{
	 font-size: 2.2em; /*what about accesibility, see larger font options */
}

.place-address{
	font-size: 0.8em;
}

.nhsuk-search__submit{
	background-color: #007f3b;
    border: 0;
    border-bottom-left-radius: 4px;
    border-bottom-right-radius: 4px;
    border-top-left-radius: 4px;
    border-top-right-radius: 4px;
    float: none;
    font-size: inherit;
    height: 52px;
    line-height: inherit;
    margin: 0;
    outline: none;
	padding: 8px 8px 0;
	margin-top: 50px;
    width: 50%;
    padding: 10px;
}

.postcode{
	box-shadow: 2px 1px #888888;
}

.label{
	font-size: 3em;
}

#map{
	position: absolute;
    height: 50%;
    width: 100%;
    z-index: 1;
}