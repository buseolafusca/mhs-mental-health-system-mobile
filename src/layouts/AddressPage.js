import React from 'react'
import { withGoogleMap, GoogleMap, Marker, InfoWindow } from "react-google-maps";
import { google } from 'google-maps';

var distance = require('google-distance-matrix');
distance.key('AIzaSyAj0bTPi8uCxzt-Fl_u9t4IHLsGpdGRwfU');
distance.units('imperial');

var NodeGeocoder = require('node-geocoder');


// var options = {
//   provider: 'google',
 
//   // Optional depending on the providers
//   httpAdapter: 'https', // Default
//   apiKey: 'AIzaSyAj0bTPi8uCxzt-Fl_u9t4IHLsGpdGRwfU', // for Mapquest, OpenCage, Google Premier
//   formatter: null         // 'gpx', 'string', ...
// };
 
// var geocoder = NodeGeocoder(options);

// geocoder.geocode('29 champs elys√©e paris', function(err, res) {
//   console.log("res");
//   console.log(res);
//   console.log("err");
//   console.log(err);
// });

// var service = new google.maps.DistanceMatrixService();

// var origin1 = new google.maps.LatLng(55.930385, -3.118425);
// var origin2 = 'Greenwich, England';
// var destinationA = 'Stockholm, Sweden';
// var destinationB = new google.maps.LatLng(50.087692, 14.421150);

// var service = new google.maps.DistanceMatrixService();
// service.getDistanceMatrix(
//   {
//     // origins: [origin1, origin2],
//     // destinations: [destinationA, destinationB],
//     origins: ['San Francisco CA'],
//     destinations: ['New York NY', '41.8337329,-87.7321554'],
//     travelMode: 'DRIVING',
//     // transitOptions: TransitOptions,
//     // drivingOptions: DrivingOptions,
//     // unitSystem: UnitSystem,
//     avoidHighways: Boolean,
//     avoidTolls: Boolean,
//   }, callback);

// function callback(response, status) {
//   console.log("response");
//   console.log(response);
//   console.log("status");
//   console.log(status);

// }


// var bounds = new google.maps.LatLngBounds;
// var markersArray = [];

// var origin1 = {lat: 55.93, lng: -3.118};
// var origin2 = 'Greenwich, England';
// var destinationA = 'Stockholm, Sweden';
// var destinationB = {lat: 50.087, lng: 14.421};

// var destinationIcon = 'https://chart.googleapis.com/chart?' +
//     'chst=d_map_pin_letter&chld=D|FF0000|000000';
// var originIcon = 'https://chart.googleapis.com/chart?' +
//     'chst=d_map_pin_letter&chld=O|FFFF00|000000';
// var map = new google.maps.Map(document.getElementById('map'), {
//   center: {lat: 55.53, lng: 9.4},
//   zoom: 10
// });

// var geocoder = new google.maps.Geocoder;


class AddressPage extends React.Component {

  componentWillMount () {
    var origins = ['San Francisco CA'];
    var destinations = ['New York NY', '41.8337329,-87.7321554'];
    
  
    distance.matrix(origins, destinations, function (err, distances) {      
      console.log("AAA");
      console.log(err);
      console.log(distances);
      
    });

  }

  render () {
    return (
      <div>sss</div>
    )
  }

}

export default AddressPage

