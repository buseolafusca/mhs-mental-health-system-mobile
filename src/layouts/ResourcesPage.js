import React from 'react'
import { getCategoriesBasedOnLocation, getListBasedOnCategoryAndLocation, getPlaceDetails } from '../services/BackendService'
import { BrowserRouter as Router, Route } from 'react-router-dom'

import '../sass/app.scss'
import '../assets/css/ResourcesPage.css'
import { Switch } from 'react-router'
import NHSHeader from '../components/NHSHeader.js'
import NHSFooter from '../components/NHSFooter.js'
import 'here-js-api/scripts/mapsjs-core'
import 'here-js-api/scripts/mapsjs-service'
import 'here-js-api/scripts/mapsjs-ui'
import 'here-js-api/scripts/mapsjs-mapevents'
import 'here-js-api/scripts/mapsjs-clustering'
import 'here-js-api/scripts/mapsjs-places'
import { appId as apI, appCode as apC } from '../variables/URLs'

class ResourcesPage extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      categoriesList: [],
      categoryListID: [],
      coordinates: []
    }
    this.handleClickofCategory = this.handleClickofCategory.bind(this)
    this.handleGoBackButton = this.handleGoBackButton.bind(this)
  }

  componentWillMount () {
    console.log("ResourcesPage")
    const { id } = this.props.match.params

    this.setState({ coordinates: id })
    var array = []
    var catID = {}
    getCategoriesBasedOnLocation(id).then(response => {
      response.data.items.forEach(element => {
        var category = element.within[0]
        if (category != null) {
          var splitedCategory = category.split('-')
          var finalCategory = ''
          splitedCategory.forEach(cat => {
            finalCategory += cat.charAt(0).toUpperCase() + cat.slice(1) + ' '
          })
          if (!array.includes(finalCategory, 0)) {
            catID[finalCategory] = category
            array.push(finalCategory)
          }
        }
      })
      this.setState({ categoriesList: array, categoryListID: catID })
    })
  }

  handleClickofCategory (item) {
    this.props.history.push('/resources/' + this.state.coordinates + '/' + this.state.categoryListID[item])
  }

  handleGoBackButton (pg) {
    this.props.history.push('/locationpage')
  }

  render () {
    return (
      <Router>
        <Switch>
          <Route path='/resources/:id' exact render={() => {
            return (
              <div class='nhsuk-expander-group'>
                <NHSHeader />
                <div class='nhsuk-back-link resourcepage-back-link'>
                  <a class='nhsuk-back-link__link resourcepage-back-link__link' onClick={() => { this.handleGoBackButton('/locationpage') }}>
                    <svg class='nhsuk-icon nhsuk-icon__chevron-left recourcepage-icon__Chevron-left' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24' aria-hidden='true'>
                      <path d='M8.5 12c0-.3.1-.5.3-.7l5-5c.4-.4 1-.4 1.4 0s.4 1 0 1.4L10.9 12l4.3 4.3c.4.4.4 1 0 1.4s-1 .4-1.4 0l-5-5c-.2-.2-.3-.4-.3-.7z' />
                    </svg>
                    Go back</a>
                </div>
                {this.state.categoriesList.map((item, key) =>
                  <details class='nhsuk-details expander' >
                    <summary class='nhsuk-details__summary' onClick={() => { this.handleClickofCategory(item) }}>
                      <span class='nhsuk-details__summary'>
                        {item}
                      </span>
                    </summary>
                  </details>
                )}

              </div>
            )
          }} />
        </Switch>
      </Router>
    )
  }
}

class PlacesPage extends React.Component {
  constructor (props) {
    // console.log("props")
    // console.log(props)
    // console.log(JSON.stringify(props))
    super(props)
    this.state = {
      placesList: [],
      coordinates: [],
      category: ''
    }
    this.handleGoBackButton = this.handleGoBackButton.bind(this)
    this.handleClickOfPlace = this.handleClickOfPlace.bind(this)
  }

  componentWillMount () {
    console.log("PlacesPage")
    const { id } = this.props.match.params
    const { cat } = this.props.match.params
    this.setState({ coordinates: id })
    this.setState({ category: cat })
    try {
      getListBasedOnCategoryAndLocation(id, cat, 1000).then(response => {
        var placesArray = []
        response.data.results.items.forEach(place => {
          placesArray.push(place)
        })
        this.setState({ placesList: placesArray })
        if (this.state.placesList.length === 0) {
          var place = {}
          place.id = '-1'
          place.title = 'No places found'
          place = [place]
          this.setState({ placesList: place })
        }
      })
    } catch (error) {
      console.log('Error in GET categories')
    }
  }

  handleGoBackButton (pg) {
    this.setState({ placesList: [] })
    this.props.history.push('/resources/' + this.state.coordinates)
  }

  handleClickOfPlace (place) {
    if (place.id !== '-1') {
      // this.props.history.push('/resources/' + this.state.coordinates + '/' + this.state.category + '/' + JSON.stringify(place));
      this.props.history.push({
        pathname: '/resources/' + this.state.coordinates + '/' + this.state.category + '/' + place.title,
        state: { detail: place }
      })
    }
  }

  render () {
    return (
      <Router>
        <Switch>
          <Route path='/resources/:id/:cat' render={() => {
            return (
              <div class='nhsuk-expander-group'>
                <NHSHeader />
                <div class='nhsuk-back-link resourcepage-back-link'>
                  <a class='nhsuk-back-link__link resourcepage-back-link__link' onClick={() => { this.handleGoBackButton('/resources/' + this.state.coordinates) }}>
                    <svg class='nhsuk-icon nhsuk-icon__chevron-left recourcepage-icon__Chevron-left' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24' aria-hidden='true'>
                      <path d='M8.5 12c0-.3.1-.5.3-.7l5-5c.4-.4 1-.4 1.4 0s.4 1 0 1.4L10.9 12l4.3 4.3c.4.4.4 1 0 1.4s-1 .4-1.4 0l-5-5c-.2-.2-.3-.4-.3-.7z' />
                    </svg>
                    Go back</a>
                </div>
                {this.state.placesList.map((item, key) =>
                  <details class='nhsuk-details expander' >
                    <summary class='nhsuk-details__summary' onClick={() => { this.handleClickOfPlace(item) }}>
                      <span class='nhsuk-details__summary'>
                        {item.title}
                      </span>
                    </summary>
                  </details>
                )}
                {/* <NHSFooter /> Footer causes problems TODO Fix */}
              </div>
            )
          }} />
        </Switch>

      </Router>
    )
  }
}

class SinglePlacePage extends React.Component {
  constructor (props) {
    console.log("33333")
    // console.log(JSON.stringify(props))
    super(props)
    console.log(props)
    console.log(props.location)
    this.state = {
      coordinates: [],
      category: '',
      place: props.location.state.detail,
      placeDetails: {},
      placeLocation: {},
      placeCoordinates: {},
      appID: apI,
      appCode: apC
    }
    if (this.state.place === '') {
      this.state.place = { title: 'MAPS API ERROR' }
    }
    this.handleGoBackButton = this.handleGoBackButton.bind(this)
    this.handleGoBackButton = this.handleGoBackButton.bind(this)
    this.interactiveMap = this.interactiveMap.bind(this)
  }

  componentWillMount () {
    console.log("SinglePlacePage")
    const { id } = this.props.match.params
    const { cat } = this.props.match.params
    // const { place } = this.props.match.params
    this.setState({ coordinates: id }) //! !!!!
    this.setState({ category: cat }) //! !!!!
    getPlaceDetails(this.state.place.href).then(response => {
      this.setState({ placeDetails: response.data })
      this.state.placeDetails = response.data
      this.setState({ placeLocation: this.state.placeDetails.location.address })
      this.setState({ placeCoordinates: this.state.placeDetails.location.position })
      // this.render();
      this.interactiveMap()
    })
  }

  handleGoBackButton (pg) {
    this.setState({ placesList: [] })
    this.props.history.push('/resources/' + this.state.coordinates + '/' + this.state.category)
  }

  interactiveMap () {
    var coordinates = {
      lat: this.state.placeCoordinates[0],
      lng: this.state.placeCoordinates[1]
    }
    var H = window.H

    /**
 * Boilerplate map initialization code starts below:
 */
    // Step 1: initialize communication with the platform
    // In your own code, replace window.app_id with your own app_id
    // and window.app_code with your own app_code
    var platform = new H.service.Platform({
      app_id: this.state.appID,
      app_code: this.state.appCode,
      useCIT: true,
      useHTTPS: true
    })

    // Step 2: initialize a map - this map is centered over Europe
    var d = document.createElement('div')
    d.id = 'map'
    document.getElementById('mapcontainer').appendChild(d)
    var defaultLayers = platform.createDefaultLayers()
    var map = new H.Map(document.getElementById('map'),
      defaultLayers.normal.map, {
        center: { lat: 50, lng: 5 },
        zoom: 4
      })
    // add a resize listener to make sure that the map occupies the whole container
    window.addEventListener('resize', () => map.getViewPort().resize())

    // Step 3: make the map interactive
    // MapEvents enables the event system
    // Behavior implements default interactions for pan/zoom (also on mobile touch environments)
    var behavior = new H.mapevents.Behavior(new H.mapevents.MapEvents(map))

    // Create the default UI components
    var ui = H.ui.UI.createDefault(map, defaultLayers)

    var placemarker = new H.map.Marker(coordinates)
    map.addObject(placemarker)
    // Now use the map as required...
    map.setCenter(coordinates)
    map.setZoom(14)
  }

  render () {
    return (
      <Router>
        <Switch>
          <Route path='/resources/:id/:cat' render={() => {
            return (
             
              <div class='nhsuk-expander-group'>
                <NHSHeader />
                <div class='nhsuk-back-link resourcepage-back-link'>
                  <a class='nhsuk-back-link__link resourcepage-back-link__link' onClick={() => { this.handleGoBackButton('/resources/' + this.state.coordinates) }}>
                    <svg class='nhsuk-icon nhsuk-icon__chevron-left recourcepage-icon__Chevron-left' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24' aria-hidden='true'>
                      <path d='M8.5 12c0-.3.1-.5.3-.7l5-5c.4-.4 1-.4 1.4 0s.4 1 0 1.4L10.9 12l4.3 4.3c.4.4.4 1 0 1.4s-1 .4-1.4 0l-5-5c-.2-.2-.3-.4-.3-.7z' />
                    </svg>
                    Go back</a>
                </div>

                <div id='placeList' class='place'>
                  <a class='place-title' href={this.state.placeDetails.view}>{this.state.placeDetails.name}</a>
                  <br />
                  <a class='place-address'>{this.state.placeLocation.house} {this.state.placeLocation.street}</a>
                  <br />
                  <a class='place-address'>{this.state.placeLocation.district}</a>
                  <br />
                  <a class='place-address'>{this.state.placeLocation.city}</a>
                  <br />
                  <div id='mapcontainer' />

                </div>
                {/* <Helmet></Helmet> */}
                <NHSFooter />
              </div>
            )
          }} />
        </Switch>

      </Router>
    )
  }
}

export { ResourcesPage, PlacesPage, SinglePlacePage }
