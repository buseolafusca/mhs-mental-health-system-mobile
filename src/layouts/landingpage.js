import React from 'react'
import '../assets/css/LandingPage.css'
import NHSHeader from '../components/NHSHeader.js'
import NHSFooter from '../components/NHSFooter.js'
import { fetchPublishedQuestionnaires } from '../services/BackendService.js'

class Square extends React.Component {
  render () {
    return (
      <a href={this.props.url} className='box' style={{ color: '#fff' }}>{this.props.title}</a>
    )
  }
}

class LandingPage extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      questionnaireList: []
    }
  }

  componentWillMount () {
    fetchPublishedQuestionnaires().then(
      response => {
        this.setState({ questionnaireList: response.questionnaireList })
      }
    )
  }

  render () {
    return (
      <div className='landing-page-container'>
        <NHSHeader />
        <div className='wrapper-container'>
          <div className='wrapper'>
            {this.state.questionnaireList.map((item, key) =>
              <Square url={'/questionnaire/' + item._id} title={item.title} key={key} />
            )}
            <Square url='/referrals' title='Referrals' />
            <Square url='/appointments' title='Appointments' />
            <Square url='/locationpage' title='Resources' />
          </div>
        </div>
        <NHSFooter />
      </div>
    )
  }
}

export default LandingPage
