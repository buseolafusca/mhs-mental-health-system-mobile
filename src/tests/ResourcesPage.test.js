import React from 'react'
import { shallow } from 'enzyme'
import { ResourcesPage, SinglePlacePage, PlacesPage } from '../layouts/ResourcesPage'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ReactDOM from 'react-dom'

// describe('<ResourcesPage />', () => {
//   it('renders three <Foo /> components', () => {
//     const match = { params: { id: 'aaa' } }
//     const wrapper = shallow(<ResourcesPage match={match} />);
//     expect(wrapper.find('nhsuk-expander-group')).to.have.lengthOf(1);
//   });


//   // it('renders an `.icon-star`', () => {
//   //   const wrapper = shallow(<MyComponent />);
//   //   expect(wrapper.find('.icon-star')).to.have.lengthOf(1);
//   // });

//   // it('renders children when passed in', () => {
//   //   const wrapper = shallow((
//   //     <MyComponent>
//   //       <div className="unique" />
//   //     </MyComponent>
//   //   ));
//   //   expect(wrapper.contains(<div className="unique" />)).to.equal(true);
//   // });

//   // it('simulates click events', () => {
//   //   const onButtonClick = sinon.spy();
//   //   const wrapper = shallow(<Foo onButtonClick={onButtonClick} />);
//   //   wrapper.find('button').simulate('click');
//   //   expect(onButtonClick).to.have.property('callCount', 1);
//   // });
// });

it('renders without crashing', () => {
  const div = document.createElement('div')
  const match = { params: { id: 'aaa' } }
  ReactDOM.render(<ResourcesPage match={match} />, div)
  ReactDOM.unmountComponentAtNode(div)
  
})


describe('ResourcesPage', () => {
  it('should render correctly in "debug" mode', async () => {
    // the id is for E1 4EY
    const match = { params: { id: 'aaa' } }
    const component = await shallow(<ResourcesPage match={match} />)
    expect(component).toMatchSnapshot()
    expect(component.find(SinglePlacePage).length).toEqual(0); 
    expect(component.state().categoriesList).toEqual([]);

  })
})


describe('PlacesPage', () => {
  it('should render correctly in "debug" mode', async () => {
    // the id is for E1 4EY
    const match = { params: {history:{length:25,action:"POP",location:{pathname:"/resources/51.51985,-0.05121/eat-drink",search:"",hash:"",key:"xdmfxd"}},location:{pathname:"/resources/51.51985,-0.05121/eat-drink",search:"",hash:"",key:"xdmfxd"},match:{path:"/resources/:id/:cat",url:"/resources/51.51985,-0.05121/eat-drink",isExact:true,params:{id:"51.51985,-0.05121",cat:"eat-drink"}}} }
    const component = await shallow(<PlacesPage match={match} />)
    expect(component).toMatchSnapshot()

  })
})

 // ResourcesPage.js |    10.53 |        0 |     9.38 |    10.64 |... 71,275,279,284 |


describe('SinglePlacePage', () => {
  it('should render correctly in "debug" mode', async () => {
    // the id is for E1 4EY
    // const match = { params: {history:{length:26,action:"PUSH",location:{pathname:"/resources/51.51985,-0.05121/eat-drink/Genesis Cinema",state:{detail:{position:[51.52078,-0.05129],distance:104,title:"Genesis Cinema",averageRating:0,category:{id:"cinema",title:"Cinema",href:"https://places.cit.api.here.com/places/v1/categories/places/cinema?app_id=nuT8ftiOYvrfFNaFEUyV&app_code=yNZIQaMP6fRuY1D8DKsuxw",type:"urn:nlp-types:category",system:"places"},icon:"https://download.vcdn.cit.data.here.com/p/d/places2_stg/icons/categories/32.icon",vicinity:"Mile End Road<br/>E1<br/>London<br/>E1 4",having:[],type:"urn:nlp-types:place",href:"https://places.cit.api.here.com/places/v1/places/826gcpvk-ef3fe1d37e5c42ca9813492f6a7cce58;context=Zmxvdy1pZD03ZjhhZDQ0OS03YzM5LTUxZTQtODJiMi1iNzQ3YTQ2NmZkNzVfMTU2NzAwMjg3MDE1Nl8wXzYwNDImcmFuaz0w?app_id=nuT8ftiOYvrfFNaFEUyV&app_code=yNZIQaMP6fRuY1D8DKsuxw",id:"826gcpvk-ef3fe1d37e5c42ca9813492f6a7cce58",openingHours:{text:"Mon-Fri: 08:00 - 22:00<br/>Sat, Sun: 10:30 - 22:00",label:"Opening hours",isOpen:true,structured:[{start:"T080000",duration:"PT14H00M",recurrence:"FREQ:DAILY;BYDAY:MO,TU,WE,TH,FR"},{start:"T103000",duration:"PT11H30M",recurrence:"FREQ:DAILY;BYDAY:SA,SU"}]},alternativeNames:[{name:"Genesis Café",language:"en"}]}},search:"",hash:"",key:"6w7zp6"}},location:{pathname:"/resources/51.51985,-0.05121/eat-drink/Genesis Cinema",state:{detail:{position:[51.52078,-0.05129],distance:104,title:"Genesis Cinema",averageRating:0,category:{id:"cinema",title:"Cinema",href:"https://places.cit.api.here.com/places/v1/categories/places/cinema?app_id=nuT8ftiOYvrfFNaFEUyV&app_code=yNZIQaMP6fRuY1D8DKsuxw",type:"urn:nlp-types:category",system:"places"},icon:"https://download.vcdn.cit.data.here.com/p/d/places2_stg/icons/categories/32.icon",vicinity:"Mile End Road<br/>E1<br/>London<br/>E1 4",having:[],type:"urn:nlp-types:place",href:"https://places.cit.api.here.com/places/v1/places/826gcpvk-ef3fe1d37e5c42ca9813492f6a7cce58;context=Zmxvdy1pZD03ZjhhZDQ0OS03YzM5LTUxZTQtODJiMi1iNzQ3YTQ2NmZkNzVfMTU2NzAwMjg3MDE1Nl8wXzYwNDImcmFuaz0w?app_id=nuT8ftiOYvrfFNaFEUyV&app_code=yNZIQaMP6fRuY1D8DKsuxw",id:"826gcpvk-ef3fe1d37e5c42ca9813492f6a7cce58",openingHours:{text:"Mon-Fri: 08:00 - 22:00<br/>Sat, Sun: 10:30 - 22:00",label:"Opening hours",isOpen:true,structured:[{start:"T080000",duration:"PT14H00M",recurrence:"FREQ:DAILY;BYDAY:MO,TU,WE,TH,FR"},{start:"T103000",duration:"PT11H30M",recurrence:"FREQ:DAILY;BYDAY:SA,SU"}]},alternativeNames:[{name:"Genesis Café",language:"en"}]}},search:"",hash:"",key:"6w7zp6"},match:{path:"/resources/:id/:cat/:place",url:"/resources/51.51985,-0.05121/eat-drink/Genesis Cinema",isExact:true,params:{id:"51.51985,-0.05121",cat:"eat-drink",place:"Genesis Cinema"}}} }
    const match = { params: {history:{length:26,action:"PUSH",location:{pathname:"/resources/51.51985,-0.05121/eat-drink/Genesis Cinema",state:{detail:{position:[51.52078,-0.05129],distance:104,title:"Genesis Cinema",averageRating:0,category:{id:"cinema",title:"Cinema",href:"https://places.cit.api.here.com/places/v1/categories/places/cinema?app_id=nuT8ftiOYvrfFNaFEUyV&app_code=yNZIQaMP6fRuY1D8DKsuxw",type:"urn:nlp-types:category",system:"places"},icon:"https://download.vcdn.cit.data.here.com/p/d/places2_stg/icons/categories/32.icon",vicinity:"Mile End Road<br/>E1<br/>London<br/>E1 4",having:[],type:"urn:nlp-types:place",href:"https://places.cit.api.here.com/places/v1/places/826gcpvk-ef3fe1d37e5c42ca9813492f6a7cce58;context=Zmxvdy1pZD03ZjhhZDQ0OS03YzM5LTUxZTQtODJiMi1iNzQ3YTQ2NmZkNzVfMTU2NzAwMjg3MDE1Nl8wXzYwNDImcmFuaz0w?app_id=nuT8ftiOYvrfFNaFEUyV&app_code=yNZIQaMP6fRuY1D8DKsuxw",id:"826gcpvk-ef3fe1d37e5c42ca9813492f6a7cce58",openingHours:{text:"Mon-Fri: 08:00 - 22:00<br/>Sat, Sun: 10:30 - 22:00",label:"Opening hours",isOpen:true,structured:[{start:"T080000",duration:"PT14H00M",recurrence:"FREQ:DAILY;BYDAY:MO,TU,WE,TH,FR"},{start:"T103000",duration:"PT11H30M",recurrence:"FREQ:DAILY;BYDAY:SA,SU"}]},alternativeNames:[{name:"Genesis Café",language:"en"}]}},search:"",hash:"",key:"6w7zp6"}},location:{pathname:"/resources/51.51985,-0.05121/eat-drink/Genesis Cinema",state:{detail:{position:[51.52078,-0.05129],distance:104,title:"Genesis Cinema",averageRating:0,category:{id:"cinema",title:"Cinema",href:"https://places.cit.api.here.com/places/v1/categories/places/cinema?app_id=nuT8ftiOYvrfFNaFEUyV&app_code=yNZIQaMP6fRuY1D8DKsuxw",type:"urn:nlp-types:category",system:"places"},icon:"https://download.vcdn.cit.data.here.com/p/d/places2_stg/icons/categories/32.icon",vicinity:"Mile End Road<br/>E1<br/>London<br/>E1 4",having:[],type:"urn:nlp-types:place",href:"https://places.cit.api.here.com/places/v1/places/826gcpvk-ef3fe1d37e5c42ca9813492f6a7cce58;context=Zmxvdy1pZD03ZjhhZDQ0OS03YzM5LTUxZTQtODJiMi1iNzQ3YTQ2NmZkNzVfMTU2NzAwMjg3MDE1Nl8wXzYwNDImcmFuaz0w?app_id=nuT8ftiOYvrfFNaFEUyV&app_code=yNZIQaMP6fRuY1D8DKsuxw",id:"826gcpvk-ef3fe1d37e5c42ca9813492f6a7cce58",openingHours:{text:"Mon-Fri: 08:00 - 22:00<br/>Sat, Sun: 10:30 - 22:00",label:"Opening hours",isOpen:true,structured:[{start:"T080000",duration:"PT14H00M",recurrence:"FREQ:DAILY;BYDAY:MO,TU,WE,TH,FR"},{start:"T103000",duration:"PT11H30M",recurrence:"FREQ:DAILY;BYDAY:SA,SU"}]},alternativeNames:[{name:"Genesis Café",language:"en"}]}},search:"",hash:"",key:"6w7zp6"},match:{path:"/resources/:id/:cat/:place",url:"/resources/51.51985,-0.05121/eat-drink/Genesis Cinema",isExact:true,params:{id:"51.51985,-0.05121",cat:"eat-drink",place:"Genesis Cinema"}}} }
    
    const location = {pathname:"/resources/51.51985,-0.05121/eat-drink/Genesis Cinema",state:{detail:{position:[51.52078,-0.05129],distance:104,title:"Genesis Cinema",averageRating:0,category:{id:"cinema",title:"Cinema",href:"https://places.cit.api.here.com/places/v1/categories/places/cinema?app_id=nuT8ftiOYvrfFNaFEUyV&app_code=yNZIQaMP6fRuY1D8DKsuxw",type:"urn:nlp-types:category",system:"places"}}}}
    // const match = {params: {"history":{"length":26,"action":"PUSH","location":{"pathname":"/resources/51.51985,-0.05121/eat-drink/Genesis Cinema","state":{"detail":{"position":[51.52078,-0.05129],"distance":104,"title":"Genesis Cinema","averageRating":0,"category":{"id":"cinema","title":"Cinema","href":"https://places.cit.api.here.com/places/v1/categories/places/cinema?app_id=nuT8ftiOYvrfFNaFEUyV&app_code=yNZIQaMP6fRuY1D8DKsuxw","type":"urn:nlp-types:category","system":"places"},"icon":"https://download.vcdn.cit.data.here.com/p/d/places2_stg/icons/categories/32.icon","vicinity":"Mile End Road<br/>E1<br/>London<br/>E1 4","having":[],"type":"urn:nlp-types:place","href":"https://places.cit.api.here.com/places/v1/places/826gcpvk-ef3fe1d37e5c42ca9813492f6a7cce58;context=Zmxvdy1pZD03ZjhhZDQ0OS03YzM5LTUxZTQtODJiMi1iNzQ3YTQ2NmZkNzVfMTU2NzAwMjg3MDE1Nl8wXzYwNDImcmFuaz0w?app_id=nuT8ftiOYvrfFNaFEUyV&app_code=yNZIQaMP6fRuY1D8DKsuxw","id":"826gcpvk-ef3fe1d37e5c42ca9813492f6a7cce58","openingHours":{"text":"Mon-Fri: 08:00 - 22:00<br/>Sat, Sun: 10:30 - 22:00","label":"Opening hours","isOpen":true,"structured":[{"start":"T080000","duration":"PT14H00M","recurrence":"FREQ:DAILY;BYDAY:MO,TU,WE,TH,FR"},{"start":"T103000","duration":"PT11H30M","recurrence":"FREQ:DAILY;BYDAY:SA,SU"}]},"alternativeNames":[{"name":"Genesis Café","language":"en"}]}},"search":"","hash":"","key":"6w7zp6"}},"location":{"pathname":"/resources/51.51985,-0.05121/eat-drink/Genesis Cinema","state":{"detail":{"position":[51.52078,-0.05129],"distance":104,"title":"Genesis Cinema","averageRating":0,"category":{"id":"cinema","title":"Cinema","href":"https://places.cit.api.here.com/places/v1/categories/places/cinema?app_id=nuT8ftiOYvrfFNaFEUyV&app_code=yNZIQaMP6fRuY1D8DKsuxw","type":"urn:nlp-types:category","system":"places"},"icon":"https://download.vcdn.cit.data.here.com/p/d/places2_stg/icons/categories/32.icon","vicinity":"Mile End Road<br/>E1<br/>London<br/>E1 4","having":[],"type":"urn:nlp-types:place","href":"https://places.cit.api.here.com/places/v1/places/826gcpvk-ef3fe1d37e5c42ca9813492f6a7cce58;context=Zmxvdy1pZD03ZjhhZDQ0OS03YzM5LTUxZTQtODJiMi1iNzQ3YTQ2NmZkNzVfMTU2NzAwMjg3MDE1Nl8wXzYwNDImcmFuaz0w?app_id=nuT8ftiOYvrfFNaFEUyV&app_code=yNZIQaMP6fRuY1D8DKsuxw","id":"826gcpvk-ef3fe1d37e5c42ca9813492f6a7cce58","openingHours":{"text":"Mon-Fri: 08:00 - 22:00<br/>Sat, Sun: 10:30 - 22:00","label":"Opening hours","isOpen":true,"structured":[{"start":"T080000","duration":"PT14H00M","recurrence":"FREQ:DAILY;BYDAY:MO,TU,WE,TH,FR"},{"start":"T103000","duration":"PT11H30M","recurrence":"FREQ:DAILY;BYDAY:SA,SU"}]},"alternativeNames":[{"name":"Genesis Café","language":"en"}]}},"search":"","hash":"","key":"6w7zp6"},"match":{"path":"/resources/:id/:cat/:place","url":"/resources/51.51985,-0.05121/eat-drink/Genesis Cinema","isExact":true,"params":{"id":"51.51985,-0.05121","cat":"eat-drink","place":"Genesis Cinema"}}}}
    const component = await shallow(<SinglePlacePage match={match} location={location}/>)
    expect(component).toMatchSnapshot()

  })
})


