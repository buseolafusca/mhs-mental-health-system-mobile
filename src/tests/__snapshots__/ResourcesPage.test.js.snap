// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourcesPage should render correctly in "debug" mode 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      exact={true}
      path="/resources/:id"
      render={[Function]}
    />
  </Switch>
</BrowserRouter>
`;
